libraries/app/application.cpp

```diff
@@ -257,24 +257,30 @@ void application_impl::new_connection( const fc::http::websocket_connection_ptr&
 
    std::string username = "*";
    std::string password = "*";
-
     // Try to extract login information from "Authorization" header if present
    std::string auth = c->get_request_header("Authorization");
-   if( boost::starts_with(auth, "Basic ") ) {
-
-      FC_ASSERT( auth.size() > 6 );
-      auto user_pass = fc::base64_decode(auth.substr(6));
-
-      std::vector<std::string> parts;
-      boost::split( parts, user_pass, boost::is_any_of(":") );
-
-      FC_ASSERT(parts.size() == 2);
-
+   if( !boost::starts_with(auth, "Basic ") )
+      login->login(username, password);
```

Comment:
`username` and `password` are still `*` at this point, right?

Reply:
Yes they are. The logic hasn't changed to the previous one, it's just restructured.

Reply:
Makes no sense, right?

Reply:
It does. When there is no Basic Auth the default login (*,*) is used. This was the same in the previous logic too.